pkg/vendir/directory/staging_dir.go

```diff
@@ -21,7 +21,10 @@ type StagingDir struct {
 }
 
 func NewStagingDir() StagingDir {
-	rootDir := ".vendir-tmp"
+	rootDir, err := os.MkdirTemp(".", ".vendir-tmp-")
+	if err != nil {
+		panic(fmt.Errorf("Creating tmp dir: %s", err))
```

Comment:
Instead of panicking, let us return an error change the function signature since it is only used in 1 place in the code.

Reply:
It calmed down 🙂